libraries/AP_HAL/system.h

```diff
@@ -28,4 +28,48 @@ uint64_t native_millis64();
 void dump_stack_trace();
 void dump_core_file();
 
+template <typename T>
+inline bool timeout_expired(const T past_time, const T now, const T timeout)
+{
+    const T dt = now - past_time;
+    return (dt > timeout);
+}
+
+template <typename T>
+inline T timeout_remaining(const T past_time, const T now, const T timeout)
+{
+    const T dt = now - past_time;
+    return (dt > timeout) ? T(0) : (timeout - dt);
```

Comment:
```
>=
```

Reply:
Fixed, great example of why this stuff is hard and easy to get wrong!